test/python/algorithms/time_evolvers/classical_methods/test_scipy_imaginary_evolver.py

```diff
@@ -0,0 +1,165 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2022.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""Test Classical Real Evolver."""
+import unittest
+from test.python.algorithms import QiskitAlgorithmsTestCase
+from ddt import data, ddt, unpack
+import numpy as np
+from qiskit.algorithms.time_evolvers.time_evolution_problem import TimeEvolutionProblem
+from qiskit.opflow import (
+    I,
+    X,
+    Z,
+    Zero,
+    Plus,
+    Minus,
+    PauliSumOp,
+    DictStateFn,
+    OperatorBase,
+    VectorStateFn,
+)
+from qiskit.quantum_info.states.statevector import Statevector
+
+from qiskit import QuantumCircuit
+from qiskit.algorithms import SciPyImaginaryEvolver
+
+
+@ddt
+class TestSciPyImaginaryEvolver(QiskitAlgorithmsTestCase):
+    """Test SciPy Imaginary Evolver."""
+
+    def create_hamiltonian_lattice(self, num_sites: int) -> PauliSumOp:
+        """Creates an ising hamiltonian on a lattice."""
+        j_const = 0.1
+        g_const = -1.0
+
+        zz_op = ["I" * i + "ZZ" + "I" * (num_sites - i - 2) for i in range(num_sites - 1)]
+        x_op = ["I" * i + "X" + "I" * (num_sites - i - 1) for i in range(num_sites)]
+        return PauliSumOp.from_list(
+            list(zip(zz_op, len(zz_op) * [j_const])) + list(zip(x_op, len(x_op) * [g_const]))
+        )
+
+    @data(
+        [Zero, 100, X, Minus],
+        [Zero, 100, -X, Plus],
```

Comment:
A minor thing but perhaps we might use tuples instead as we don't modify the content.